cache/models/samples.js

```diff
@@ -0,0 +1,155 @@
+
+module.exports = {
+
+  /**
+   * Retrieves the sample from redis and sends it back in the response.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  getSampleFromRedis(req, res, next) {
+    const resultObj = { reqStartTime: new Date() }; // for logging
+    const sampleName = req.swagger.params.key.value.toLowerCase();
+    const aspectName = sampleName.split('|')[ONE];
+    const commands = [];
+
+    // get sample
+    commands.push([
+      'hgetall',
+      sampleStore.toKey(constants.objectType.sample, sampleName),
+    ]);
+
+    // get aspect
+    commands.push([
+      'hgetall',
+      sampleStore.toKey(constants.objectType.aspect, aspectName),
+    ]);
+
+    redisClient.batch(commands).execAsync()
+    .then((responses) => {
+      resultObj.dbTime = new Date() - resultObj.reqStartTime; // log db time
+
+      // clean and attach aspect to sample
+      const sampleRes = cleanAddAspectToSample(
+        responses[ZERO], responses[ONE], res.method
+      );
+
+      u.logAPI(req, resultObj, sampleRes); // audit log
+      res.status(httpStatus.OK).json(sampleRes);
+    })
+    .catch((err) => u.handleError(next, err, helper.modelName));
+  },
+
+  /**
+   * Finds zero or more samples from redis and sends them back in the response.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  findSamplesFromRedis(req, res, next) {
+    const resultObj = { reqStartTime: new Date() }; // for logging
+    const sampleCmds = [];
+    const aspectCmds = [];
+    const response = [];
+
+    // get all Samples
+    redisClient.smembersAsync(constants.indexKey.sample)
+    .then((allSampleKeys) => {
+      // add to commands to get sample
+      allSampleKeys.forEach((sampleName) => {
+        sampleCmds.push(
+          ['hgetall', sampleName.toLowerCase()]
+        );
+      });
+
+      // get all aspect names
+      return redisClient.smembersAsync(constants.indexKey.aspect);
+    })
+    .then((allAspectKeys) => {
+      // add to commands to get aspect
+      allAspectKeys.forEach((aspectName) => {
+        aspectCmds.push(
+          ['hgetall', aspectName.toLowerCase()]
+        );
+      });
+
+      // get all samples and aspects
+      return Promise.all([
+        redisClient.batch(sampleCmds).execAsync(),
+        redisClient.batch(aspectCmds).execAsync(),
+      ]);
+    })
+    .then((sampleAndAspects) => {
+      resultObj.dbTime = new Date() - resultObj.reqStartTime; // log db time
+      const samples = sampleAndAspects[ZERO];
+      const aspects = sampleAndAspects[ONE];
+
+      samples.forEach((sampleObj) => {
+        const sampleAspect = aspects.find((aspect) =>
+          aspect.name === sampleObj.name.split('|')[ONE]
+        );
+
```

Comment:
I guess for sample endpoints, it would not matter much because its not used that often as compared to hierarchy endpoint.